### PR TITLE
 bug fix for building with vs 2013 on windows OS 

### PR DESCRIPTION
**Pull request by json87**

### CODE REVIEW

File: src/base/cost_functions.h [hunk 1]
```diff
@@ -89,58 +89,69 @@ class BundleAdjustmentCostFunction {
 // camera calibration and point parameters, and fixed camera pose.
 template <typename CameraModel>
 class BundleAdjustmentConstantPoseCostFunction {
- public:
```
[the remark this answers is not in the document]
Comment (json87) [reply src/base/cost_functions.h]: when building with vs 2013, the initialization of "qvec_" and "tvec_" would also cause building errors.

File: src/exe/colmap.cc [hunk 2]
```diff
@@ -1559,23 +1563,36 @@ int RunVocabTreeMatcher(int argc, char** argv) {
 
 std::vector<Image> ReadVocabTreeRetrievalImageList(const std::string& path,
                                                    Database* database) {
-  std::vector<Image> images;
```
[the remark this answers is not in the document]
Comment (json87) [reply src/exe/colmap.cc]: in my test, I found that the ReadImageWithName function does not work properly.

File: src/base/cost_functions.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/base/cost_functions.h]: What I meant is the formatting of the code. Can you please run clang-format on the file? Or make sure you don't change unrelated lines.

[the remark this answers is not in the document]
Comment (json87) [reply src/base/cost_functions.h]: ok, I get it and revert them to clang format

**Comment by ahojnnes** [comment]
Thanks!

**Pull request merged**